### ISSUE TITLE
can't find system header files in c++ built-in headers and local c headers

### ISSUE THREAD

**Issue by WJ430981** [clangd/clangd#1243]
I use cmake to generate the compile_commands.json,
every thing goes well, but when i open built-in c++ header files or  header files, there's wrong with can't find the system headers.
clangd detect the right header include path, from  `Driver produce` log, only local C++ source, header and local C source file get the right path to parse. I tryied `.clangd`  or `compile_flags.txt` to configure the compile flags, failed.

Env:
```
host:  Windows10
edit: vscode
clangd: 14.0.3
vscode clangd 0.1.21
toolchain:  aarch64-none-linux-gnu from developer.arm.com
```

compile_commands.json
```
[
{
  "directory": "E:/git/HybridModule/build.vscode",
  "command": "D:\\Developer\\aarch64-none-linux-gnu\\bin\\aarch64-none-linux-gnu-gcc.exe -IE:/git/HybridModule/include -IE:/git/HybridModule/src -g -o CMakeFiles\\c_module.dir\\src\\c_module\\c_module.c.obj -c E:\\git\\HybridModule\\src\\c_module\\c_module.c",
  "file": "E:\\git\\HybridModule\\src\\c_module\\c_module.c"
},
{
  "directory": "E:/git/HybridModule/build.vscode",
  "command": "D:\\Developer\\aarch64-none-linux-gnu\\bin\\aarch64-none-linux-gnu-g++.exe -IE:/git/HybridModule/include -IE:/git/HybridModule/src -g -o CMakeFiles\\cxx_module.dir\\src\\cxx_module\\cxx_module.cpp.obj -c E:\\git\\HybridModule\\src\\cxx_module\\cxx_module.cpp",
  "file": "E:\\git\\HybridModule\\src\\cxx_module\\cxx_module.cpp"
},
{
  "directory": "E:/git/HybridModule/build.vscode",
  "command": "D:\\Developer\\aarch64-none-linux-gnu\\bin\\aarch64-none-linux-gnu-g++.exe -IE:/git/HybridModule/include -IE:/git/HybridModule/src -g -o CMakeFiles\\test.dir\\test\\test.cpp.obj -c E:\\git\\HybridModule\\test\\test.cpp",
  "file": "E:\\git\\HybridModule\\test\\test.cpp"
}
]
```

c++ built-in headers
![image](https://user-images.githubusercontent.com/8575414/183588397-76fc0e2f-274e-4ddf-90f0-faed6f83ce9c.png)

local c headers
![image](https://user-images.githubusercontent.com/8575414/183588258-6ea77339-e8e3-4dfc-a0d0-29be95557504.png)

Driver produce log for local c header file
![image](https://user-images.githubusercontent.com/8575414/183590070-072004ec-0f48-4cbc-a22d-0bbbe8b954c9.png)

[clangd14.0.3_with_aarch64-none-linux-gnu.txt](https://github.com/clangd/clangd/files/9288181/clangd14.0.3_with_aarch64-none-linux-gnu.txt)

**Comment by HighCommander4** [comment]
I suspect you are running into #1100, which is fixed on 15.x / trunk.

**Comment by WJ430981** [comment]
awesome, fixed with [15.0.0-rc2](https://github.com/llvm/llvm-project/releases/tag/llvmorg-15.0.0-rc2) from here.
